apps/google-docs/src/components/page/GoogleDocUploader.tsx

```diff
@@ -0,0 +1,164 @@
+import { useState, useEffect } from 'react';
+import { Box, Button, Form, FormControl, TextInput } from '@contentful/f36-components';
+import { PageAppSDK } from '@contentful/app-sdk';
+
+interface GoogleDocUploaderProps {
+  sdk: PageAppSDK;
+  onSuccess: (title: string, html: string | null) => void;
+  onError: (message: string) => void;
+  isDisabled?: boolean;
+}
+
+function isValidGoogleDocUrl(url: string): boolean {
+  return /^https:\/\/docs\.google\.com\/document\/d\/[A-Za-z0-9_-]+\/edit(?:\?[^#]*)?$/.test(url);
```

Comment:
Here's a couple small (and totally optional) regex simplifications that can be made here:

- matching the ID `[A-Za-z0-9_-]+` -> `[\w-]+`

- matching the optional query string `(?:\?[^#]*)?$` -> `(?:\?.*)?$`